### ISSUE TITLE
Bad URLs can cause app to crash

### ISSUE THREAD

**Issue by ssilvert** [keycloak/keycloak-admin-ui#1157]
We have a general problem where a bad URL can cause the app to crash.  The only way to recover is to reload the entire application.

For instance, go to http://localhost:8080/#/master/clients/FOOBAR/settings

Note that `FOOBAR` replaces the client id.

You will get an error that says, `Cannot destructure property 'redirectUris' of 'client2' as it is null.`  The retry link does not help.  

This happens for many entities, and some can even happen in the course of normal usage.  For another example:

1. Go to Realm Settings --> Events --> Admin events settings
2. Turn on `Save events`
3. Go to Groups and create a group.
4. Then delete the group
5. Go to Events --> Admin events
6. Click the link to the resource path of the deleted group
7. You will get a similar error as the FOOBAR example.

The bottom line is that we need a general way to handle bad URLs that does not crash the applicaiton.

**Comment by edewit** [comment]
@ssilvert it's not crashing it's the error page that you see the message is what went wrong. In this case there is no logic for when the client is not found don't think we want a generic error for that we just need logic to deal with that on these screens

**Comment by ssilvert** [comment]
@edewit To the user, it's a crash.  The application can no longer be used until you reload.  And the user doesn't know he needs to refresh his browser to get it working again.

Of course, if we can catch the error and handle it properly then we should do that on a case-by-case basis.

But in the generic case, if there is no way to catch this type of error in a way that lets you continue then at least we need to provide a better error page and a link that says something like "Press here to refresh and continue".

**Comment by edewit** [comment]
In this particular case refreshing the browser or clicking the "Retry" / "Press here to refresh and continue" will not work as it will just give you the same error.

**Comment by ssilvert** [comment]
@edewit What if it was treated as an external link?  So say I access the admin console from localhost:8080.  If the error page has this, it should work:

`<a href="http://localhost:8080/auth/admin/master/console/#/">Click here to reload</a>`
